ui/pages/permissions-connect/flask/snap-update/snap-update.js

```diff
@@ -85,8 +105,19 @@ export default function SnapUpdate({
           variant={TypographyVariant.H7}
           as="span"
         >
-          {t('snapUpdateExplanation', [`${request.metadata.dappOrigin}`])}
+          {t('snapUpdateExplanation', [`${request.metadata?.dappOrigin}`])}
```

Comment:
How come this can be undefined now?

Reply:
I removed the transition step that rendered another page between all the approvals. that means that some of the data can take some time to arrive to the approvals.

Reply:
But specifically `request.metadata` should always be defined, shouldn't it? 🤔

Reply:
Yes but when everything unmounts we loose that data